artifactory/commands/utils/result_test.go

```diff
@@ -0,0 +1,96 @@
+package utils
+
+import (
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/tests"
+	clientutils "github.com/jfrog/jfrog-client-go/utils"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+// Check a case that targetRepository is not written in a deployableArtifacts file and needs to be read from the config file.
+func TestUnmarshalDeployableArtifacts(t *testing.T) {
+	err, cleanUpJfrogHome := CreateTempJfrogConfig()
+	defer cleanUpJfrogHome()
+	// DeployableArtifact file is being change during runtime so a copy needs to be created.
+	tempDeployableArtifacts, err := createTempDeployableArtifactFile()
+	defer os.Remove(tempDeployableArtifacts)
+	r, err := UnmarshalDeployableArtifacts(path.Join(tempDeployableArtifacts), path.Join(getTestsDataGradlePath(), "config", "gradle.yaml"))
+	assert.NoError(t, err)
+	for transferDetails := new(clientutils.FileTransferDetails); r.reader.NextRecord(transferDetails) == nil; transferDetails = new(clientutils.FileTransferDetails) {
+		assert.True(t, strings.HasPrefix(transferDetails.TargetPath, "http://localhost:8080/artifactory/"))
+		assert.True(t, strings.Contains(transferDetails.TargetPath, "gradle-local-repo"))
+	}
+}
+
+func createTempDeployableArtifactFile() (string, error) {
+	testsDataGradlePath := getTestsDataGradlePath()
+	summary, err := os.Open(path.Join(testsDataGradlePath, "deployableArtifacts", "summary"))
+	tmpSummary, err := os.Create(path.Join(testsDataGradlePath, "deployableArtifacts", "tmpSummary"))
```

Comment:
We're not supposed to add or modify files under the source code directory. That includes the testdata dir. We should create all the temp files under one directory. The code should use one locations (and API to get it).

In addition, jfrog-client-go already includes a ```CopyFile``` function. Let's reuse it instead of creating a new one.